app/src/main/java/com/github/se/travelpouch/model/ListTravelViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package com.github.se.travelpouch.model
+
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+open class ListTravelViewModel(private val repository: TravelRepository) : ViewModel() {
+  private val travels_ = MutableStateFlow<List<TravelContainer>>(emptyList())
+  val travels: StateFlow<List<TravelContainer>> = travels_.asStateFlow()
+
+  private val selectedTravel_ = MutableStateFlow<TravelContainer?>(null)
+  open val selectedTravel: StateFlow<TravelContainer?> = selectedTravel_.asStateFlow()
+
+  init {
+    repository.init { getTravels() }
+  }
+
+  /**
+   * Generates a new unique ID.
+   *
+   * @return A new unique ID.
+   */
+  fun getNewUid(): String {
+    return repository.getNewUid()
+  }
+
+  /** Gets all Travel documents. */
+  fun getTravels() {
+    repository.getTravels(onSuccess = { travels_.value = it }, onFailure = {})
```

Comment:
It might be a good idea to add proper error handling for all methods to avoid silent failures like this :

onFailure = { e -> 
    // Handle or log error
    Log.e("ListTravelViewModel", "Failed to retrieve travels", e) 
}